design/rook-ceph-status-conditions.md

```diff
@@ -0,0 +1,93 @@
+	   Failure 		: If any failure occurs in the Ceph Cluster
+	   Cluster Expanding	: If the Cluster is Expanding
+
+## Implementation Details
+
+Implementation as Specified in https://github.com/openshift/custom-resource-status/:
+
+We can add Conditions simply in the Custom Resource struct as:
+	  
+	   type ClusterStatus struct{
+		Conditions     *RookConditions    `json:"conditions,omitempty"`
+		CephStatus     *CephStatus        `json:"ceph,omitempty"`
+	   }
+
+After that we can just make changes inside rook ceph codebase as necessary. 
+	   
+The definition of the type Conditions can have the following details:
+	   
+	   Type               RookConditionType  `json:"type" description:"type of Rook condition"`
+  	   Status             ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`
+  	   Reason             *string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
+  	   Message            *string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+	   LastHeartbeatTime  *unversioned.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
+	   LastTransitionTime *unversioned.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transition from one status to another"`
+
+The fields `Reason`, `Message`, `LastHeartbeatTime`, `LastTransitionTime` are optional field. Though the use of `Reason` field is encouraged.
+
+Condition Types field specifies the current state of the system. Condition status values may be `True`, `False`, or `Unknown`. The absence of a condition should be interpreted the same as Unknown. How controllers handle Unknown depends on the Condition in question.
+`Reason` is intended to be a one-word, CamelCase representation of the category of cause of the current status, and `Message` is intended to be a human-readable phrase or sentence, which may contain specific details of the individual occurrence. `Reason` is intended to be used in concise output, such as one-line kubectl get output, and in summarizing occurrences of causes, whereas `Message` is intended to be presented to users in detailed status explanations, such as `kubectl describe output`.
+
+In the CephClusterStatus, we can either remove the `Status.State` and `Status.Message` fields and call the `Conditions` structure from inside the `ClusterStatus`, or we can just add the `Conditions` structure by keeping the already included fields.The first method is preferred because the `Conditions` structure contains the `Conditions.Type` and `Conditions.Message` which is similar to the `Status.State` and `Status.Message`. According to the above changes, necessary changes are to be made everywhere `ClusterStatus` or one of its fields are referred. 
+
+### Examples
+Example 1
```

Comment:
Instead of just writing "Example 1", can you add a small summary about what this example is illustrating? 

"RookConditions" would be a array with multiple "States". Can you try incorporating few of those? Like, when cluster is "Progressing", what would the Status of "Upgradable" be? What would be Status of "Available"? (During every reconcile if you set status to Progressing, does it mean it is not Available?)

Reply:
During the "Progressing" the cluster is only getting created right? So, the Upgradable status is not yet added to the RookConditon array. But once the Cluster is in "Available" state and then if it goes into Upgrading, then the status of the Available will be False, as well as the status of Ready.

@umangachapagain Also is there any difference between the Ready State and Available state?